features/mbedtls/targets/TARGET_Cypress/TARGET_PSOC6/mbedtls_device.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2019 Cypress Semiconductor Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBEDTLS_DEVICE_H
+#define MBEDTLS_DEVICE_H
+
+/* Currently this target supports SHA1 */
+// #define MBEDTLS_SHA1_C
+
+#define MBEDTLS_SHA1_ALT
+#define MBEDTLS_SHA256_ALT
+#define MBEDTLS_SHA512_ALT
+
+/* Currently this target supports CBC, CFB, OFB, CTR and XTS cipher modes */
+#define MBEDTLS_AES_ALT
+// #define MBEDTLS_CIPHER_MODE_CBC
+// #define MBEDTLS_CIPHER_MODE_CFB
+// #define MBEDTLS_CIPHER_MODE_OFB
+// #define MBEDTLS_CIPHER_MODE_CTR
+// #define MBEDTLS_CIPHER_MODE_XTS
```

Comment:
Which block modes have been tested? Which block modes are supported?

Reply:
ECB, CBC, CFB, CTR are supported and tested

Reply:
Could you please test XTS?

Reply:
Yes, will do this for the next release.

Reply:
Please add `#error` for when building your ALT implementation. I believe it'll get stuck at run time, but it's better to fail at compile time than at run time.

Reply:
@Patater, XTS implementation included into AES ALT also successfully tested by mbedtls green-tea test suite

Reply:
@Patater, sounds like there is no need to add #error for XTS mode in this case.

Reply:
If it works, yeah, great. I expected the two AES contexts each to reserve the AES hardware and not be usable at the same time. How does the sharing work?